### DISCUSSION TITLE
DAP configuration

### DISCUSSION THREAD

**Discussion by Je12emy** [mason-org/mason.nvim#218] [Q&A] [answered]
How do I configure a DAP for a given language? Previously, I had the following configuration for `netcoredb`

```lua
dap.adapters.coreclr = {
    type = 'executable',
    command = '/home/foo/netcoredbg/netcoredbg',
    args = {'--interpreter=vscode'}
}

dap.configurations.cs = {
    {
        type = "coreclr",
        name = "launch - netcoredbg",
        request = "launch",
        program = function()
            return vim.fn.input('Path to dll', vim.fn.getcwd() .. '/bin/Debug/',
                                'file')
        end
    }
}
```

How do I port this configuration to work through mason? I did not find anything related in the docs

**Comment by williamboman** [comment, marked as answer]
Hello! So Mason doesn't replace any existing editor functionalities, it just provides you with a way to manage the installation of `netcoredbg`. From the snippet you provided, the only thing I'd change is the `command`:

```diff
dap.adapters.coreclr = {
    type = 'executable',
-   command = '/home/foo/netcoredbg/netcoredbg',
+   command = 'netcoredbg',
    args = {'--interpreter=vscode'}
}
```

**Reply by Je12emy** [reply]
Awesome, thank you so much for this project!

**Reply by nyngwang** [reply]
@williamboman So the `command` will pick the local one `netcorebg` right?

**Reply by williamboman** [reply]
It’ll pick the first netcoredbg it finds in PATH, which should be mason’s unless you’ve changed default behaviour or altered PATH manually after setting up mason
